docs/guides/writing_stubs.rst

```diff
@@ -337,6 +337,45 @@ No::
     DAY_FLAG: int
     NIGHT_FLAG: int
 
+Overloads
+---------
+
+All variants of overloaded functions and methods must have an ``@overload``
+decorator. Do not include the implementation's final non-`@overload`-decorated
+definition.
+
+Yes::
+
+  @overload
+  def foo(x: str) -> str: ...
+  @overload
+  def foo(x: float) -> int: ...
+
+No::
+
+  @overload
+  def foo(x: str) -> str: ...
+  @overload
+  def foo(x: float) -> int: ...
+  def foo(x: str | float) -> Any: ...
+
+Decorators
+----------
+
+Include only those decorators whose effects type checkers understand, enumerated
```

Comment:
Maybe it’d be better to describe these as decorators that have “special behaviour”, since type checkers do understand other decorators via typeshed? With the current list the distinction seems more that these are special-cased, and their results can’t be described by regular types. And you want to apply the decorator results where possible to avoid relying on checker-specific inference behaviour in your library.

Reply:
Thanks for the feedback!

> type checkers do understand other decorators via typeshed

This isn't true for all type checkers. Pytype, for example, ignores any decorators in stubs outside of this list. I tweaked this sentence to say "all of the major type checkers" rather than just "type checkers," since I think that makes it a little clearer that the concern here is differences in behavior among type checkers.